### PR TITLE
Fix issue with DJI O3 showing miles instead of kFt

### LINKED ISSUE #9005: INAV 6.1 RC1 OSD Altitude Shown in Miles > 1000 ft
## Current Behavior
Using INAV 6.1 RC1 with the OSD set to imperial units, altitude displays in feet below 1000 feet and in miles over 1000 feet. I have observed this in both the relative altitude display above your home location and the absolute MSL altitude display.
## Steps to Reproduce

1. Flash INAV 6.1 RC1
2. Set OSD to imperial units and turn on both relative and MSL altitude. My OSD mode was BF43COMPAT with a DJI O3 system and DJI Goggles V2, but I don't think this is unique to that.
3. Fly the model to over 1000 feet, or launch from an MSL altitude over 1000 feet.
4. Altitude displays are in miles over 1000 feet altitude.

## Expected behavior
 Altitude should always display in feet when using imperial units. This is correct in INAV 6.0.

## Suggested solution(s)
Change all OSD altitude displays to always display in feet with imperial units. Only lateral distance should switch to miles.

## Additional context

set osd_video_system = BF43COMPAT
set osd_units = IMPERIAL
set osd_time_alarm = 0
set osd_alt_alarm = 0
set osd_dist_alarm = 0
set osd_neg_alt_alarm = 0
set osd_link_quality_alarm = 50
set osd_rssi_dbm_alarm = -100
set osd_switch_indicators_align_left = OFF

osd_layout 0 0 27 2 H
osd_layout 0 1 0 14 V
osd_layout 0 2 0 0 V
osd_layout 0 3 8 6 V
osd_layout 0 4 8 6 V
osd_layout 0 6 24 14 V
osd_layout 0 7 14 13 V
osd_layout 0 9 14 15 V
osd_layout 0 11 7 15 V
osd_layout 0 12 5 14 V
osd_layout 0 13 3 8 H
osd_layout 0 14 0 1 V
osd_layout 0 15 23 8 V
osd_layout 0 20 12 0 V
osd_layout 0 21 1 0 V
osd_layout 0 22 15 2 V
osd_layout 0 23 0 2 V
osd_layout 0 24 6 1 V
osd_layout 0 26 24 11 V
osd_layout 0 28 24 15 H
osd_layout 0 29 16 2 H
osd_layout 0 30 2 3 V
osd_layout 0 34 11 1 V
osd_layout 0 37 0 12 H
osd_layout 0 38 0 15 V
osd_layout 0 40 24 13 V
osd_layout 0 43 0 0 V
osd_layout 0 85 3 8 V
osd_layout 0 96 25 6 V
osd_layout 0 100 14 14 H
osd_layout 0 108 17 2 H
osd_layout 0 109 24 0 V
osd_layout 0 110 26 1 V
osd_layout 0 111 25 4 H
osd_layout 0 112 25 2 V
osd_layout 0 128 0 11 V

---

- FC Board name and vendor: INAV/MATEKF405TE_SD 6.1.0 Apr 21 2023 / 17:48:27 (2d7a057d)
- INAV version string: GCC-10.2.1 20201103 (release)

### PR DESCRIPTION
**Pull request by MrD-RC**
Fixes #9005

Tested on other systems via HITL. All fine. Should be tested by someone with the O3 system for final confirmation.

### CODE REVIEW

**Comment by prelator** [comment]
Looks like this corrects it to give altitude in thousands of feet. While better, would it be possible to just always give altitude in feet without decimals, like INAV 6.0 does?

File: src/main/io/osd.c [hunk 1]
```diff
@@ -547,25 +557,25 @@ void osdFormatAltitudeSymbol(char *buff, int32_t alt)
         case OSD_UNIT_IMPERIAL:
             if (osdFormatCentiNumber(buff + 4 - digits, CENTIMETERS_TO_CENTIFEET(alt), 1000, 0, 2, digits)) {
                 // Scaled to kft
-                buff[4] = SYM_ALT_KFT;
+                buff[symbolIndex++] = symbolKFt;
```
Comment (prelator) [inline src/main/io/osd.c:560]: Could this KFt formatting be removed completely? I think most people using imperial units would prefer to see altitude always in feet without decimals. Ie. 1500 ft. vs 1.500.

Comment (MrD-RC) [reply src/main/io/osd.c:560]: No, due to space. There is only room for 3 digits. INAV has been kFt for years. 1500 ft would be 1.50 kFt.

File: src/main/io/osd.c [hunk 2]
```diff
@@ -532,13 +532,23 @@ static void osdFormatWindSpeedStr(char *buff, int32_t ws, bool isValid)
...
+    uint8_t digits = 4U;
+    uint8_t symbolIndex = 4U;
+    uint8_t symbolKFt = SYM_ALT_KFT;
+
+    if (alt >= 0) {
+        digits = 3U;
         buff[0] = ' ';
     }
+
+#ifndef DISABLE_MSP_BF_COMPAT   // IF BFCOMPAT is not supported, there's no need to check for it and change the values
```
Comment (mmosca) [inline src/main/io/osd.c:544]: Probably not needed. In this case IsBfCompat is a constant macro and hopefully the if gets optimized away, but aldo doesn't hurt.

**Comment by MrD-RC** [comment]
> Looks like this corrects it to give altitude in thousands of feet. While better, would it be possible to just always give altitude in feet without decimals, like INAV 6.0 does?

INAV 6 doesn't do that. Altitude has always been kFt. If 6.0 showed something different, it was because of this bug. The altitude has space only for 3 digits. Which is why kFt is used.

File: src/main/io/osd.c (hunk 2 above)
Comment (MrD-RC) [reply src/main/io/osd.c:544]: Hopefully it will all be unnecessary in the future. Are you able to test this on the O3 system?

**Comment by mmosca** [comment]
I can confirm it will whos 1.XX above 1000ft.

**Comment by MrD-RC** [comment]
Thanks @mmosca 

ready to merge in that case.

**Pull request merged**